extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -287,6 +287,12 @@ export class AmpSlideScroll extends BaseSlides {
   }
 
   /** @override */
+  unlayoutCallback() {
+    this.slideIndex_ = null;
```

Comment:
@camelburrito I have added the unit test "should NOT call showSlide_ before re-layout". I checked that: 

- The unit test fails without the change in `slidescroll.js` (i.e. the previous situation).
- The unit test succeeds with the fix applied.

Hope it's ready to be approved now.

Thanks,
Peter